src/Raven.Server/Documents/Indexes/MapReduce/Static/MapReduceIndex.cs

```diff
@@ -18,7 +18,7 @@ namespace Raven.Server.Documents.Indexes.MapReduce.Static
 {
     public class MapReduceIndex : MapReduceIndexBase<MapReduceIndexDefinition>
     {
-        private readonly HashSet<CollectionName> _referencedCollections = new HashSet<CollectionName>();
+        private readonly HashSet<CollectionName> _referencedCollections = new HashSet<CollectionName>(CollectionNameComparer.Instance);
```

Comment:
What I'm saying, is that this is created once per index (very rare) and is never used otherwise. There is no perf benefit here.